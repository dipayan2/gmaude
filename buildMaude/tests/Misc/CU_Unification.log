PASS CU_Unification (exit status: 0)
