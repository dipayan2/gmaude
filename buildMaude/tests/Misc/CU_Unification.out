==========================================
unify in FOO : f(X, Y) =? f(U, V) .

Unifier 1
X --> #1:Foo
Y --> #2:Foo
U --> #1:Foo
V --> #2:Foo

Unifier 2
X --> 1f
Y --> #1:Foo
U --> 1f
V --> #1:Foo

Unifier 3
X --> 1f
Y --> f(#1:Foo, #2:Foo)
U --> #1:Foo
V --> #2:Foo

Unifier 4
X --> 1f
Y --> #1:Foo
U --> 1f
V --> #1:Foo

Unifier 5
X --> #1:Foo
Y --> #2:Foo
U --> 1f
V --> f(#1:Foo, #2:Foo)
==========================================
variant unify in FOO : f(X, Y) =? f(U, V) .

Unifier 1
rewrites: 0
X --> %1:Foo
Y --> %2:Foo
U --> %1:Foo
V --> %2:Foo

Unifier 2
rewrites: 0
X --> 1f
Y --> f(%1:Foo, %2:Foo)
U --> %1:Foo
V --> %2:Foo

Unifier 3
rewrites: 0
X --> %1:Foo
Y --> %2:Foo
U --> 1f
V --> f(%1:Foo, %2:Foo)

No more unifiers.
rewrites: 0
==========================================
unify in FOO : f(X, Y) =? X .

Unifier 1
X --> 1f
Y --> 1f
==========================================
variant unify in FOO : f(X, Y) =? X .

Unifier 1
rewrites: 0
X --> 1f
Y --> 1f

No more unifiers.
rewrites: 0
==========================================
unify in FOO : f(X, Y) =? Y .

Unifier 1
X --> 1f
Y --> #1:Foo
==========================================
variant unify in FOO : f(X, Y) =? Y .

Unifier 1
rewrites: 0
X --> 1f
Y --> %1:Foo

No more unifiers.
rewrites: 0
==========================================
unify in FOO2 : f(X, Y) =? f(U, V) .

Unifier 1
X --> #1:Foo
Y --> #2:Foo
U --> #1:Foo
V --> #2:Foo

Unifier 2
X --> #1:Foo
Y --> 1f
U --> #1:Foo
V --> 1f

Unifier 3
X --> f(#1:Foo, #2:Foo)
Y --> 1f
U --> #1:Foo
V --> #2:Foo

Unifier 4
X --> #1:Foo
Y --> 1f
U --> #1:Foo
V --> 1f

Unifier 5
X --> #1:Foo
Y --> #2:Foo
U --> f(#1:Foo, #2:Foo)
V --> 1f
==========================================
variant unify in FOO2 : f(X, Y) =? f(U, V) .

Unifier 1
rewrites: 0
X --> %1:Foo
Y --> %2:Foo
U --> %1:Foo
V --> %2:Foo

Unifier 2
rewrites: 0
X --> f(%1:Foo, %2:Foo)
Y --> 1f
U --> %1:Foo
V --> %2:Foo

Unifier 3
rewrites: 0
X --> %1:Foo
Y --> %2:Foo
U --> f(%1:Foo, %2:Foo)
V --> 1f

No more unifiers.
rewrites: 0
==========================================
unify in FOO2 : f(X, Y) =? X .

Unifier 1
X --> #1:Foo
Y --> 1f
==========================================
variant unify in FOO2 : f(X, Y) =? X .

Unifier 1
rewrites: 0
X --> %1:Foo
Y --> 1f

No more unifiers.
rewrites: 0
==========================================
unify in FOO2 : f(X, Y) =? Y .

Unifier 1
X --> 1f
Y --> 1f
==========================================
variant unify in FOO2 : f(X, Y) =? Y .

Unifier 1
rewrites: 0
X --> 1f
Y --> 1f

No more unifiers.
rewrites: 0
==========================================
unify in FOO3 : f(X, Y) =? f(U, V) .

Unifier 1
X --> #1:Foo
Y --> #2:Foo
U --> #1:Foo
V --> #2:Foo

Unifier 2
X --> 1f
Y --> #1:Foo
U --> 1f
V --> #1:Foo

Unifier 3
X --> 1f
Y --> #1:Foo
U --> #1:Foo
V --> 1f

Unifier 4
X --> 1f
Y --> f(#1:Foo, #2:Foo)
U --> #1:Foo
V --> #2:Foo

Unifier 5
X --> #1:Foo
Y --> 1f
U --> 1f
V --> #1:Foo

Unifier 6
X --> #1:Foo
Y --> 1f
U --> #1:Foo
V --> 1f

Unifier 7
X --> f(#1:Foo, #2:Foo)
Y --> 1f
U --> #1:Foo
V --> #2:Foo

Unifier 8
X --> 1f
Y --> #1:Foo
U --> 1f
V --> #1:Foo

Unifier 9
X --> #1:Foo
Y --> 1f
U --> 1f
V --> #1:Foo

Unifier 10
X --> #1:Foo
Y --> #2:Foo
U --> 1f
V --> f(#1:Foo, #2:Foo)

Unifier 11
X --> 1f
Y --> #1:Foo
U --> #1:Foo
V --> 1f

Unifier 12
X --> #1:Foo
Y --> 1f
U --> #1:Foo
V --> 1f

Unifier 13
X --> #1:Foo
Y --> #2:Foo
U --> f(#1:Foo, #2:Foo)
V --> 1f
==========================================
variant unify in FOO3 : f(X, Y) =? f(U, V) .

Unifier 1
rewrites: 0
X --> %1:Foo
Y --> %2:Foo
U --> %1:Foo
V --> %2:Foo

Unifier 2
rewrites: 0
X --> 1f
Y --> f(%1:Foo, %2:Foo)
U --> %1:Foo
V --> %2:Foo

Unifier 3
rewrites: 0
X --> f(%1:Foo, %2:Foo)
Y --> 1f
U --> %1:Foo
V --> %2:Foo

Unifier 4
rewrites: 0
X --> %1:Foo
Y --> %2:Foo
U --> 1f
V --> f(%1:Foo, %2:Foo)

Unifier 5
rewrites: 0
X --> %1:Foo
Y --> %2:Foo
U --> f(%1:Foo, %2:Foo)
V --> 1f

No more unifiers.
rewrites: 0
==========================================
unify in FOO3 : f(X, Y) =? X .

Unifier 1
X --> 1f
Y --> 1f

Unifier 2
X --> #1:Foo
Y --> 1f
==========================================
variant unify in FOO3 : f(X, Y) =? X .

Unifier 1
rewrites: 0
X --> %1:Foo
Y --> 1f

No more unifiers.
rewrites: 0
==========================================
unify in FOO3 : f(X, Y) =? Y .

Unifier 1
X --> 1f
Y --> #1:Foo

Unifier 2
X --> 1f
Y --> 1f
==========================================
variant unify in FOO3 : f(X, Y) =? Y .

Unifier 1
rewrites: 0
X --> 1f
Y --> %1:Foo

No more unifiers.
rewrites: 0
==========================================
unify in FOO4 : f(X, Y) =? f(U, V) .

Unifier 1
X --> #1:Foo
Y --> #2:Foo
V --> #1:Foo
U --> #2:Foo

Unifier 2
X --> #2:Foo
Y --> #1:Foo
V --> #1:Foo
U --> #2:Foo

Unifier 3
X --> 1f
Y --> #1:Foo
V --> 1f
U --> #1:Foo

Unifier 4
X --> 1f
Y --> #1:Foo
V --> #1:Foo
U --> 1f

Unifier 5
X --> 1f
Y --> f(#1:Foo, #2:Foo)
V --> #1:Foo
U --> #2:Foo

Unifier 6
X --> #1:Foo
Y --> 1f
V --> 1f
U --> #1:Foo

Unifier 7
X --> #1:Foo
Y --> 1f
V --> #1:Foo
U --> 1f

Unifier 8
X --> f(#1:Foo, #2:Foo)
Y --> 1f
V --> #1:Foo
U --> #2:Foo

Unifier 9
X --> 1f
Y --> #1:Foo
V --> 1f
U --> #1:Foo

Unifier 10
X --> #1:Foo
Y --> 1f
V --> 1f
U --> #1:Foo

Unifier 11
X --> #1:Foo
Y --> #2:Foo
V --> 1f
U --> f(#1:Foo, #2:Foo)

Unifier 12
X --> 1f
Y --> #1:Foo
V --> #1:Foo
U --> 1f

Unifier 13
X --> #1:Foo
Y --> 1f
V --> #1:Foo
U --> 1f

Unifier 14
X --> #1:Foo
Y --> #2:Foo
V --> f(#1:Foo, #2:Foo)
U --> 1f
==========================================
variant unify in FOO4 : f(X, Y) =? f(U, V) .

Unifier 1
rewrites: 0
X --> %1:Foo
Y --> %2:Foo
V --> %1:Foo
U --> %2:Foo

Unifier 2
rewrites: 0
X --> %2:Foo
Y --> %1:Foo
V --> %1:Foo
U --> %2:Foo

Unifier 3
rewrites: 0
X --> 1f
Y --> f(%1:Foo, %2:Foo)
V --> %1:Foo
U --> %2:Foo

Unifier 4
rewrites: 0
X --> f(%1:Foo, %2:Foo)
Y --> 1f
V --> %1:Foo
U --> %2:Foo

Unifier 5
rewrites: 0
X --> %1:Foo
Y --> %2:Foo
V --> 1f
U --> f(%1:Foo, %2:Foo)

Unifier 6
rewrites: 0
X --> %1:Foo
Y --> %2:Foo
V --> f(%1:Foo, %2:Foo)
U --> 1f

No more unifiers.
rewrites: 0
==========================================
unify in FOO4 : f(X, Y) =? X .

Unifier 1
X --> 1f
Y --> 1f

Unifier 2
X --> #1:Foo
Y --> 1f
==========================================
variant unify in FOO4 : f(X, Y) =? X .

Unifier 1
rewrites: 0
X --> %1:Foo
Y --> 1f

No more unifiers.
rewrites: 0
==========================================
unify in FOO4 : f(X, Y) =? Y .

Unifier 1
X --> 1f
Y --> #1:Foo

Unifier 2
X --> 1f
Y --> 1f
==========================================
variant unify in FOO4 : f(X, Y) =? Y .

Unifier 1
rewrites: 0
X --> 1f
Y --> %1:Foo

No more unifiers.
rewrites: 0
Bye.
